content-review/impl/dao/src/main/java/org/sakaiproject/contentreview/dao/ContentReviewItemDao.java

```diff
@@ -77,6 +77,16 @@
 		
 		return Optional.ofNullable((ContentReviewItem) c.uniqueResult());
 	}
+	
+	public Optional<ContentReviewItem> findByProviderAndExternalId(Integer providerId, String externalId) {
+
+		Criteria c = sessionFactory.getCurrentSession()
+				.createCriteria(ContentReviewItem.class)
+				.add(Restrictions.eq("providerId", providerId))
+				.add(Restrictions.eq("externalId", externalId));
+		
+		return Optional.ofNullable((ContentReviewItem) c.uniqueResult());
```

Comment:
did you test this function to see if it works?

Reply:
yep